src/DataReader/parse_queries.rs

```diff
@@ -172,6 +175,50 @@ fn build_expression_from_pair(pair: pest::iterators::Pair<Rule>) -> QueryExpress
     }
 }
 
+fn build_state_from_pair(pair: pest::iterators::Pair<Rule>) -> QueryExpression {
+    let mut inner_pair = pair.clone().into_inner();
+    let locPair = inner_pair.next().unwrap();
+
+    let mut loc_names:Vec<Box<QueryExpression>> = Vec::new();
+    for loc_name in locPair.as_str().split(','){
+        loc_names.push(Box::new(QueryExpression::LocName(loc_name.trim().to_string())));
+    }
+
+    let clockPair = inner_pair.next().unwrap();
+
+    let invariantVersion:Option<Box<BoolExpression>> = 
+    if clockPair.as_str().trim() != "" {
+        let clockString = clockPair.as_str().trim().to_string().replace(",", "&&");
+        let invariantVersion = match parse(&clockString) {
+            Ok(result) => result,
+            Err(result) => panic!("{:?}", result),
+        };
+        Some(Box::new(invariantVersion))
+    }
+    else { None };
```

Comment:
Maybe add a comment explaining that we are calling the invariant parser

Reply:
Fix